Eevee.Sleep.Bot/Handlers/InteractionHandler.cs

```diff
@@ -116,4 +119,25 @@ await interaction
                 throw new ArgumentException($"Unhandled modal ID: {modalId}");
         }
     }
+
+    private static async Task OnButtonClicked(SocketMessageComponent component)
+    {
+        var customId = component.Data.CustomId;
+        if (!customId.StartsWith("role"))
+        {
+            return;
+        }
+
+        await component.RespondAsync("You have been given the role!", ephemeral: true);
+
+        var roleId = ulong.Parse(customId.Replace("role", ""));
```

Comment:
For the interaction info (de)serialization, make them a class and put all (de)serialization impl there

Reply:
Already answered here
https://github.com/RaenonX-PokemonSleep/eevee-sleep-discord-bot/pull/25#discussion_r1609964437